src/main/java/com/gregtechceu/gtceu/common/machine/multiblock/electric/DistillationTowerMachine.java

```diff
@@ -0,0 +1,358 @@
+                                if (firstValid == null) firstValid = (IFluidHandler) h;
+                            },
+                                    () -> fluidOutputs.add(VoidFluidHandler.INSTANCE));
+                    outputIndex++;
+                } else if (part.self().getPos().getY() > y) {
+                    fluidOutputs.add(VoidFluidHandler.INSTANCE);
+                } else {
+                    GTCEu.LOGGER.error(
+                            "The Distillation Tower at {} has a fluid export hatch with an unexpected Y position",
+                            getPos());
+                    onStructureInvalid();
+                    return;
+                }
+            }
+        }
+    }
+
+    @Override
+    public void onStructureInvalid() {
+        fluidOutputs = null;
+        firstValid = null;
+        super.onStructureInvalid();
+    }
+
+    public int limitParallel(GTRecipe recipe, int multiplier) {
+        int minMultiplier = 0;
+        int maxMultiplier = multiplier;
+
+        var maxAmount = recipe.getOutputContents(FluidRecipeCapability.CAP).stream()
+                .map(Content::getContent)
+                .map(FluidRecipeCapability.CAP::of)
+                .filter(i -> !i.isEmpty())
+                .map(i -> i.getStacks()[0])
+                .mapToInt(FluidStack::getAmount)
+                .max()
+                .orElse(0);
+
+        if (maxAmount == 0) return multiplier;
+
+        while (minMultiplier != maxMultiplier) {
+            if (multiplier > Integer.MAX_VALUE / maxAmount) multiplier = Integer.MAX_VALUE / maxAmount;
+            GTRecipe copy = recipe.copy(ContentModifier.multiplier(multiplier), false);
+            boolean filled = getRecipeLogic().applyFluidOutputs(copy, FluidAction.SIMULATE);
+            int[] bin = ParallelLogic.adjustMultiplier(filled, minMultiplier, multiplier, maxMultiplier);
+            minMultiplier = bin[0];
+            multiplier = bin[1];
+            maxMultiplier = bin[2];
+        }
+        return multiplier;
+    }
+
+    public static class DistillationTowerLogic extends RecipeLogic {
+
+        @Nullable
+        @Persisted
+        @DescSynced
+        GTRecipe workingRecipe = null;
+
+        public DistillationTowerLogic(IRecipeLogicMachine machine) {
+            super(machine);
+        }
+
+        @NotNull
+        @Override
+        public DistillationTowerMachine getMachine() {
+            return (DistillationTowerMachine) super.getMachine();
+        }
+
+        // Copy of lastRecipe with fluid outputs trimmed, for output displays like Jade or GUI text
+        @Override
+        public @Nullable GTRecipe getLastRecipe() {
+            return workingRecipe;
+        }
+
+        @Override
+        @Nullable
+        public Iterator<GTRecipe> searchRecipe() {
+            var recipeType = machine.getRecipeType();
+            if (recipeType == GTRecipeTypes.DISTILLERY_RECIPES) return super.searchRecipe();
```

Comment:
Might be able to merge this together with the interface I suggested for the parallel limit

Reply:
Needs to be hard coded in order to use DTs output matching method 
[`GTRecipeType#searchRecipe`](https://github.com/GregTechCEu/GregTech-Modern/blob/c64bd1d549a4484e7194a5333a6fc3ced361d164/src/main/java/com/gregtechceu/gtceu/api/recipe/GTRecipeType.java#L220) , which is used in the default recipe logic `searchRecipe`, is hardcoded to use `GTrecipe#matchRecipe`. This does not work for me because it would not respect the separation of layers. 

In a future PR, I plan to extract out this repeated hardcoded matching condition from various places in RecipeLogic so that it becomes easier to write custom recipe logic without copy pasting a lot of methods with only a single line change (like you see here with the DT logic). However that is beyond the scope of this PR.